amethyst_assets/src/lib.rs

```diff
@@ -6,6 +6,18 @@
 //! * extensibility
 //! * asynchronous & parallel using rayon
 //! * allow different sources
+//!
+//! # Overview
+//!
+//! The most important type of this crate is the `AssetStorage`.
+//! It's the place where all the assets are located; it only gives you
+//! `Handle`s to them which you can use to get a reference to the actual asset.
+//! The `Loader` is responsible for loading asset data, an intermediate format in asset loading.
+//! Asset data will then be pushed to a queue which the `AssetStorage` has access to.
+//! After that, we of course need to transform asset data into an asset, which happens
+//! by calling `AssetStorage::process`. After the data is processed, the handle will
```

Comment:
I should add that this is typically done by systems and that there's a default implementation (`Processor`) for it.